src/main/java/com/amazon/opendistroforelasticsearch/sql/query/ESActionFactory.java

```diff
@@ -98,10 +98,18 @@ else if(isJoin(sqlExpr,sql)){
                 IndexStatement describeStatement = new IndexStatement(StatementType.DESCRIBE, sql);
                 return new DescribeQueryAction(client, describeStatement);
             default:
-                throw new SQLFeatureNotSupportedException(String.format("Unsupported query: %s", sql));
+                throw new SQLFeatureNotSupportedException(
+                        String.format("Query must start with SELECT, DELETE, SHOW or DESCRIBE: %s", sql));
         }
     }
 
+    private static String getFirstWord(String sql) {
```

Comment:
would be good to somehow document (either a comment, or an assert that the first character is not a space) that there's a precondition to this method : the string is already trimmed. Otherwise calling with a string `" SELECT f1, f2 FROM ..."` will return the whole string.

Reply:
This is a private method in a class that has only one entry point and the method is called right on top of it. I feel that your suggestion might be an overkill

Reply:
Sounds good, no objections.